bench_test.go

```diff
@@ -232,15 +234,20 @@ func CalcStatsJsoniterWithPath(str string) (s Stats) {
 	return
 }
 
+// strToBytes returns byte slice from string without allocations.
+func strToBytes(s string) []byte {
+	return unsafe.Slice((*byte)(unsafe.Pointer((*reflect.StringHeader)(unsafe.Pointer(&s)).Data)), len(s))
```

Comment:
I totally get your point and it sure is a bit unfair to have the extra overhead because of a different input type but I'm still not sure how I feel about introducing `unsafe` to the party since usually it's not welcome in production code 😕

Reply:
Since this library is focusing on processing strings - it's fair to compare only _legal and safe_ methods of use for the context of this library only. If any user of a library requires performance at the cost of the risks of using `unsafe` then it's his/her responsibility to do so.

However, **we can let it pass** under the assumption that the user of [go-faster/jx](https://github.com/go-faster/jx) is willing to pay that price (which many folks often do) just in order to eliminate the overhead for a fair comparison's sake.

Reply:
A bit of a background:

I actually started out with `unsafe` but quickly encountered a problem on my M1 Max macOS 12.1 Go 1.17.6. Seemingly random bugs started to appear after converting `string` to `[]byte` using the `unsafe` method from [valyala/fasthttp](https://github.com/valyala/fasthttp/blob/e9db537178708982f64736dbdedfddd2a168d395/bytesconv.go#L333). I couldn't reproduce them neither on another Intel i7 macOS 12.1 Go 1.17.6 MBP, nor on [go.dev/play](https://go.dev/play). **Only later did I realize that it's me who did the stupid mistake** of using an unsafe `[]byte -> string` method for `string -> []byte` conversion, which, IMHO, only proves the dangers and cost of utilizing `unsafe`.

Reply:
I agree that it is unsafe, but not sure how to compare libraries fairly, because string -> byte allocation is unrelated.

If you will find another way I'm ok with it!

Reply:
Yes, it's unrelated and slightly falsifies the results.
The solution is to just provide a new function:
```go
func Scan(o Options, s []byte, fn func(*IteratorBytes) (err bool)) Error
```
This will be the most widely used function anyway.